### DISCUSSION TITLE
classes/class_javascriptbridge

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#124] [User-contributed notes]
# classes/class_javascriptbridge

Inherits: Object Singleton that connects the engine with the browser's JavaScript context in Web export. Description: The JavaScriptBridge singleton is implemented only in the Web export. It's used...

https://docs.godotengine.org/en/stable/classes/class_javascriptbridge.html

**Comment by ALiwoto** [comment]
Can someone please explain, how does a pwa understands *when* exactly it needs an update when calling `pwa_needs_update` function?

**Comment by stesproject** [comment]
Is this bridge enough to implement AirConsole APIs into a Godot game?

**Comment by infernostars** [comment]
I wish this part of `create_callback` was more prominent in the documentation:

> The reference must be kept until the callback happens, or it won't be called at all. 

This confused me for a while since it doesn't provide any errors and just does nothing instead, so it'd be nice if it was more apparent and not directly in the middle of the function documentation where it's likely glanced over.
